roxctl/common/k8s_port_forward.go

```diff
@@ -191,27 +205,59 @@ func runPortForward(ctx context.Context) (uint16, error) {
 	case <-readyChannel:
 	case err := <-errChan:
 		if err != nil {
-			return 0, err
+			return "", 0, err
 		}
 	}
 	ports, err := forwarder.GetPorts()
 	if err != nil {
-		return 0, errors.WithMessage(err, "failed to acquire forwarding ports")
+		return "", 0, errors.WithMessage(err, "failed to acquire forwarding ports")
 	}
-	return ports[0].Local, nil
+	// centralPort is the central pod port, not service port, as forwaring goes
+	// directly to the pod, and service name is used to pass TLS validation.
+	centralEndpoint := fmt.Sprintf("central.%s:%d", namespace, centralPort)
+	return centralEndpoint, ports[0].Local, nil
 }
 
-// GetForwardingEndpoint starts port-forwarding to svc/central and returns
-// the endpoint with local port forwarding to the service port.
-func GetForwardingEndpoint() (string, error) {
+// GetForwardingEndpoint starts port-forwarding to a svc/central pod,
+// and returns the service endpoint, to which the requests should be sent,
+// and the local endpoint forwarded to the pod, which the dialer should dial to.
+func GetForwardingEndpoint() (string, string, error) {
 	portForwardOnce.Do(func() {
 		ctx, cancel := context.WithTimeout(context.Background(), reasonableTimeout)
 		defer cancel()
-		if port, err := runPortForward(ctx); err != nil {
+		if svc, port, err := runPortForward(ctx); err != nil {
 			portForwardingError = errPortForwarding.CausedBy(err)
 		} else {
-			forwardEndpoint = fmt.Sprintf("localhost:%d", port)
+			serviceEndpoint = svc // central.stackrox:8443
+			localhostEndpoint = fmt.Sprintf("127.0.0.1:%d", port)
 		}
 	})
-	return forwardEndpoint, portForwardingError
+	return serviceEndpoint, localhostEndpoint, portForwardingError
+}
+
+// getForwardingDialContext returns a dialer, that resolves central service
+// endpoint to the localhost forwarded endpoint: this way the TLS certificate
+// validation work.
+func getForwardingDialContext() ([]byte, func(ctx context.Context, addr string) (net.Conn, error), error) {
+	_, core, namespace, err := getConfigs()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ca, err := getCentralCA(context.Background(), core, namespace)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
Currently we fail closed when the central CA cannot be fetched, however we should fail open in case folks do not have the correct RBAC (e.g. read secrets in the central namespace); they should still make use of the `use-current-k8s-context` flag.